tenets/codelingo/k8/well-named-interface/codelingo.yaml

```diff
@@ -0,0 +1,30 @@
+funcs:
+  - name: beginsWith
+    type: asserter
+    body: |
+      function (start, str) {
+        return str.indexOf(start) == 0
+      }
+tenets:
+  - name: well-named-interface
+    flows:
+      codelingo/docs:
+        title: Well Named Interface
+        body: |
+          Please consider the package name when selecting an interface name, and avoid redundancy. e.g. `storage.Interface` is better than `storage.StorageInterface`.
+      codelingo/review:
+        comment: Interface name {{interfaceName}} has name of package {{packageName} redundantly prefixed.
+    query: |
+      import codelingo/ast/go
+            
+      go.file(depth = any):
```

Comment:
Don't think you need to specify go.file

Reply:
This one needs go.file: because when followed by go.ident: it is a package name

Reply:
That being said, it's only going to match when there is an interface_type. As you mentioned next, an interface is only going to be at the top level, so technically, this may not need the go.file: due of the interface which follows. However, I feel like removing this go.file: would make the tenet non-migratable if in future there was a package definition fact and relying on side-effects that may change in future seems nasty.